src/rand.rs

```diff
@@ -109,13 +113,43 @@ impl<T> RandomlyConstructable for T where T: self::sealed::RandomlyConstructable
 /// `fill()` once at a non-latency-sensitive time to minimize latency for
 /// future calls.
 #[derive(Clone, Debug)]
-pub struct SystemRandom(());
+pub struct SystemRandom;
 
 impl SystemRandom {
     /// Constructs a new `SystemRandom`.
     #[inline(always)]
-    pub fn new() -> Self {
-        Self(())
+    pub const fn new() -> Self {
+        Self
+    }
+}
+
+#[cfg(feature = "rand-api")]
+impl RngCore for SystemRandom {
```

Comment:
It is intentional that we do not implement `rand` traits. If somebody is implementing cryptography they shouldn't use `rand`. If they aren't implementing cryptography then they probably shouldn't be using `ring::rand`. If they are implementing something that maybe is cryptography and maybe not then we'll leave it up to them as to how to solve that.

Reply:
I see- I thought there might be a reason. Thank you for your time.